poetry/console/commands/command.py

```diff
@@ -11,9 +12,17 @@
 class Command(BaseCommand):
     loggers = []
 
+    _poetry: Optional["Poetry"] = None
+
     @property
     def poetry(self) -> "Poetry":
-        return self.get_application().poetry
+        if self._poetry is None:
+            return self.get_application().poetry
+
+        return self._poetry
+
+    def set_poetry(self, poetry: "Poetry") -> None:
```

Comment:
Should this be a property setter?

Reply:
I prefer to have explicit setters for the public API. It feels clearer this way.